File: MA/flutter_lab/lib/domain/assignment.dart
class Assignment {
  late int id;
  late String name;
  late String subject;
  late DateTime dueDate;
  late DateTime assignedDate;
  late bool isCompleted = false;

  Assignment(
      {required this.id,
      required this.name,
      required this.subject,
      required this.dueDate,
      required this.assignedDate,
      required this.isCompleted});
}

File: MA/server/backend/backend/Domain/Assignment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Domain;

[Table("Assignment")]
public class Assignment
{
    [Key]
    public int Id { get; set; }
    public string Subject { get; set; } = null!;
    public string Name { get; set; } = null!;
    public DateTime DueDate { get; set; }
    public DateTime ReceivedDate { get; set; }
    public bool IsCompleted { get; set; } = false;
}